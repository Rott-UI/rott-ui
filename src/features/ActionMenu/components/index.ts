export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
