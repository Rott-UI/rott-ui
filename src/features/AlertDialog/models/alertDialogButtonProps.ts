// Util and Lib Imports
import {Variant} from '../../models'
import {TranslationLanguageTypes} from '../../../utils'

export interface AlertDialogButtonProps {
  variant?: Variant
  text: TranslationLanguageTypes
  onPress?: () => void
}
