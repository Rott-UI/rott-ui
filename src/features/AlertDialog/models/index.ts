export * from './alertDialogButtonProps'
export * from './alertDialogContextModel'
export * from './alertDialogModel'
