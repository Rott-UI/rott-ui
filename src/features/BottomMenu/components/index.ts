export * from './BottomMenu'
export * from './BottomMenuItem'
