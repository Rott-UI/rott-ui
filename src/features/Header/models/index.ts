export * from './HeaderIconProps'
export * from './HeaderProps'
