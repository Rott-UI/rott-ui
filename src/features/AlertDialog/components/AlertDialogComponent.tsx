// React Imports
import React, {FC} from 'react'

// Component Imports
import {AlertDialog, Item, Label, ModalProps, Pressable, Separator} from '../..'
import {AlertDialogButtonProps} from '../models'
import {AlertDialogStyles} from '../styles'

// Constant Imports
import {COLOURS} from '../../../constants'

// Util and Lib Imports
import {display, formatMessage} from '../../../utils'
import {ActivityIndicator} from 'react-native'

interface AlertDialogProps
  extends Omit<
    ModalProps,
    | 'header'
    | 'headerTitle'
    | 'fullScreen'
    | 'headerLogo'
    | 'closeButton'
    | 'height'
    | 'backgroundColor'
    | 'panResponderBackgroundColor'
    | 'headerBackgroundColor'
    | 'slideToClose'
  > {
  id?: number
  title: string
  text?: string
  buttons?: {
    cancelButton?: AlertDialogButtonProps
    confirmButton?: AlertDialogButtonProps
  }
  showActivityIndicator?: boolean
}

export const AlertDialogComponent: FC<AlertDialogProps> = ({
  id,
  title,
  text,
  buttons = {
    cancelButton: {
      text: 'COMMON.CANCEL',
      variant: 'danger',
      onPress: undefined,
    },
    confirmButton: undefined,
  },
  showActivityIndicator,
  ...props
}) => {
  return (
    <Item
      testID='alert-dialog-test-id'
      alignItemsCenter
      justifyContentCenter
      width={display.setWidth(80)}
      borderRadius={16}
      backgroundColor={COLOURS.GREY100}
      style={AlertDialogStyles(props).InformationModalView}>
      {showActivityIndicator && (
        <Item marginTop={24}>
          <ActivityIndicator size='large' color={COLOURS.PRIMARY} />
        </Item>
      )}

      <Item justifyContentCenter alignItemsCenter paddingHorizontal={16} paddingTop={24}>
        <Label variant='grey-900' textCenter fontSize='xl' fontWeight={600}>
          {title}
        </Label>

        <Label variant='grey-900' marginTop='5%' textCenter fontSize='md' fontWeight='normal'>
          {text}
        </Label>
      </Item>

      <Separator
        opacity={0.2}
        marginTop={24}
        orientation='horizontal'
        variant='neutral-alpha-300'
        size='full'
      />

      <Item
        row
        size='full'
        backgroundColor={COLOURS.GREY100}
        height={50}
        borderBottomStartRadius={16}
        borderBottomEndRadius={16}>
        {buttons?.cancelButton && (
          <Pressable
            testID='alert-dialog-cancel-button-test-id'
            height={50}
            justifyContentCenter
            alignItemsCenter
            onPress={() => {
              !!buttons?.cancelButton?.onPress && buttons.cancelButton.onPress()
              AlertDialog.hide(id)
            }}>
            <Item>
              <Label
                variant={buttons?.cancelButton?.variant || 'danger'}
                fontWeight={500}
                textCenter>
                {formatMessage(buttons.cancelButton.text)}
              </Label>
            </Item>
          </Pressable>
        )}

        {buttons?.confirmButton && (
          <Separator opacity={0.2} orientation='vertical' size='full' variant='neutral-alpha-300' />
        )}

        {buttons?.confirmButton && (
          <Pressable
            testID='alert-dialog-confirm-button-test-id'
            height={50}
            justifyContentCenter
            alignItemsCenter
            onPress={() => !!buttons?.confirmButton?.onPress && buttons?.confirmButton?.onPress()}>
            <Label
              variant={buttons.confirmButton.variant || 'grey-100'}
              fontWeight={500}
              textCenter>
              {formatMessage(buttons.confirmButton.text)}
            </Label>
          </Pressable>
        )}
      </Item>
    </Item>
  )
}
