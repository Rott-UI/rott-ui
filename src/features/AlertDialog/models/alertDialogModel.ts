// Model Imports
import {AlertDialogButtonProps} from './alertDialogButtonProps'

export interface AlertDialogModel {
  id?: number
  title: string
  text?: string
  showActivityIndicator?: boolean
  buttons?: {
    cancelButton?: AlertDialogButtonProps
    confirmButton?: AlertDialogButtonProps
  }
}
