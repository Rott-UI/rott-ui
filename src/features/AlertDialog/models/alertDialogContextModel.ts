// Model Imports
import {AlertDialogModel} from './alertDialogModel'

export interface AlertDialogContextModel {
  show: (alertDialogToRender: AlertDialogModel) => void
  hide: (id?: number) => void
  test: () => void
}
