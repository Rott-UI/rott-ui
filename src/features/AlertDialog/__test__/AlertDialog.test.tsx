// React Imports
import React from 'react'

// Alert Dialog Imports
import {AlertDialogComponent} from '../components'
import {AlertDialogButtonProps} from '../models'

//Util and Lib Imports
import {fireEvent, formatMessage, render} from '../../../utils'

// Mock the required props
const mockProps = {
  title: formatMessage('TEST.WITH.PARAM', {testText: 'Test Title'}),
  text: formatMessage('TEST.WITH.PARAM', {testText: 'Test Text'}),
  buttons: {
    confirmButton: {
      text: 'COMMON.OK',
      onPress: jest.fn(),
    } as AlertDialogButtonProps,
    cancelButton: {
      text: 'COMMON.CANCEL',
      onPress: jest.fn(),
    } as AlertDialogButtonProps,
  },
}

describe('AlertDialog -> Component', () => {
  const testId = {
    alertDialogTestId: 'alert-dialog-test-id',
  }

  it('alert dialog ilk render anında snapshot ile eşleşmeli', () => {
    const {alertDialogTestId} = testId

    const renderedAlertDialog = render(
      <AlertDialogComponent visible testID={alertDialogTestId} {...mockProps} />
    )

    expect(renderedAlertDialog).toMatchSnapshot()
  })

  it('alert dialog doğru renderlanmalı', () => {
    const {alertDialogTestId} = testId
    const {getByText} = render(
      <AlertDialogComponent visible testID={alertDialogTestId} {...mockProps} />
    )
    const title = getByText(formatMessage('TEST.WITH.PARAM', {testText: 'Test Title'}))
    const text = getByText(formatMessage('TEST.WITH.PARAM', {testText: 'Test Text'}))
    const confirmButton = getByText(formatMessage('COMMON.OK'))
    const cancelButton = getByText(formatMessage('COMMON.CANCEL'))

    expect(title).toBeOnTheScreen()
    expect(text).toBeOnTheScreen()
    expect(confirmButton).toBeOnTheScreen()
    expect(cancelButton).toBeOnTheScreen()
  })

  it('buttons property dolu olmadığı zaman cancel buttonu default olarak ekranda renderlanmalı', () => {
    const {alertDialogTestId} = testId
    const {title, text} = mockProps
    const {getByText} = render(
      <AlertDialogComponent visible testID={alertDialogTestId} title={title} text={text} />
    )

    const cancelButton = getByText(formatMessage('COMMON.CANCEL'))

    expect(cancelButton).toBeOnTheScreen()
  })

  it('butona tıklandığında onPress ve onClose fonksiyonları çağırılmalı', () => {
    const {alertDialogTestId} = testId
    const {
      buttons: {
        confirmButton: {onPress},
      },
    } = mockProps
    const {getByText} = render(
      <AlertDialogComponent visible testID={alertDialogTestId} {...mockProps} />
    )

    const confirmButton = getByText(formatMessage('COMMON.OK'))
    fireEvent.press(confirmButton)

    expect(onPress).toHaveBeenCalled()
  })
})
