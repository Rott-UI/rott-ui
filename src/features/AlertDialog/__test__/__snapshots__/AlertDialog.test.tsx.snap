// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertDialog -> Component alert dialog ilk render anında snapshot ile eşleşmeli 1`] = `
[
  <View
    alignItemsCenter={true}
    backgroundColor="rgba(234, 239, 240, 1)"
    borderRadius={16}
    justifyContentCenter={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(234, 239, 240, 1)",
        "bottom": undefined,
        "elevation": 5,
        "flex": undefined,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": "center",
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
        "top": undefined,
        "width": 600,
        "zIndex": undefined,
      }
    }
    testID="alert-dialog-test-id"
    width={312}
  >
    <View
      alignItemsCenter={true}
      justifyContentCenter={true}
      paddingHorizontal={16}
      paddingTop={24}
      style={
        {
          "alignItems": "center",
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": undefined,
          "justifyContent": "center",
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": 31,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": 46,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": undefined,
            "bottom": undefined,
            "color": "rgba(34, 63, 70, 1)",
            "flexGrow": undefined,
            "fontFamily": "Markpro-Medium",
            "fontSize": 18,
            "fontWeight": 600,
            "gap": undefined,
            "height": undefined,
            "left": undefined,
            "letterSpacing": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "textAlign": "center",
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
      >
        TEST WITH Test Title
      </Text>
      <Text
        marginTop="5%"
        style={
          {
            "backgroundColor": undefined,
            "bottom": undefined,
            "color": "rgba(34, 63, 70, 1)",
            "flexGrow": undefined,
            "fontFamily": "Markpro-Medium",
            "fontSize": 14,
            "fontWeight": "normal",
            "gap": undefined,
            "height": undefined,
            "left": undefined,
            "letterSpacing": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": "5%",
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "textAlign": "center",
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
      >
        TEST WITH Test Text
      </Text>
    </View>
    <View
      marginTop={24}
      style={
        {
          "backgroundColor": "rgba(34, 63, 70, 0.40)",
          "bottom": undefined,
          "flexGrow": undefined,
          "gap": undefined,
          "height": 2,
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": 46,
          "maxHeight": 2,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "opacity": 0.2,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": "100%",
          "zIndex": undefined,
        }
      }
      variant="neutral-alpha-300"
    />
    <View
      backgroundColor="rgba(234, 239, 240, 1)"
      borderBottomEndRadius={16}
      borderBottomStartRadius={16}
      height={50}
      style={
        {
          "alignItems": undefined,
          "backgroundColor": "rgba(234, 239, 240, 1)",
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "row",
          "flexGrow": undefined,
          "gap": undefined,
          "height": 96,
          "justifyContent": undefined,
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": 96,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": "100%",
          "zIndex": undefined,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        alignItemsCenter={true}
        animated={false}
        collapsable={false}
        focusable={true}
        height={50}
        justifyContentCenter={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": 96,
            "justifyContent": "center",
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": 96,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
        testID="alert-dialog-cancel-button-test-id"
      >
        <View
          style={
            {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "bottom": undefined,
              "flex": undefined,
              "flexDirection": "column",
              "flexGrow": undefined,
              "gap": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "left": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "maxHeight": undefined,
              "maxWidth": undefined,
              "minHeight": undefined,
              "minWidth": undefined,
              "overflow": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "paddingVertical": undefined,
              "position": undefined,
              "right": undefined,
              "top": undefined,
              "width": undefined,
              "zIndex": undefined,
            }
          }
        >
          <Text
            style={
              {
                "backgroundColor": undefined,
                "bottom": undefined,
                "color": "rgba(246, 83, 83, 1)",
                "flexGrow": undefined,
                "fontFamily": "Markpro-Book",
                "fontSize": 14,
                "fontWeight": 500,
                "gap": undefined,
                "height": undefined,
                "left": undefined,
                "letterSpacing": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "maxHeight": undefined,
                "maxWidth": undefined,
                "minHeight": undefined,
                "minWidth": undefined,
                "overflow": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "paddingVertical": undefined,
                "position": undefined,
                "right": undefined,
                "textAlign": "center",
                "top": undefined,
                "width": undefined,
                "zIndex": undefined,
              }
            }
          >
            İptal
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "rgba(34, 63, 70, 0.40)",
            "bottom": undefined,
            "flexGrow": undefined,
            "gap": undefined,
            "height": undefined,
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "opacity": 0.2,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": 2,
            "zIndex": undefined,
          }
        }
        variant="neutral-alpha-300"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        alignItemsCenter={true}
        animated={false}
        collapsable={false}
        focusable={true}
        height={50}
        justifyContentCenter={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": 1,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": 96,
            "justifyContent": "center",
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": 96,
            "maxWidth": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": undefined,
            "zIndex": undefined,
          }
        }
        testID="alert-dialog-confirm-button-test-id"
      >
        <Text
          style={
            {
              "backgroundColor": undefined,
              "bottom": undefined,
              "color": "rgba(234, 239, 240, 1)",
              "flexGrow": undefined,
              "fontFamily": "Markpro-Book",
              "fontSize": 14,
              "fontWeight": 500,
              "gap": undefined,
              "height": undefined,
              "left": undefined,
              "letterSpacing": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "maxHeight": undefined,
              "maxWidth": undefined,
              "minHeight": undefined,
              "minWidth": undefined,
              "overflow": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "paddingVertical": undefined,
              "position": undefined,
              "right": undefined,
              "textAlign": "center",
              "top": undefined,
              "width": undefined,
              "zIndex": undefined,
            }
          }
        >
          Tamam
        </Text>
      </View>
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
]
`;
